### ISSUE TITLE
Deep Exploitablity Decent: Convergence

### ISSUE THREAD

**Issue by dennisjay** [deepmind/open_spiel#129]
Hi there,
don't know if this is the right place to ask this but yet I don't have any other Contact Details so I'm trying it this way.

Yesterday I got an idea about an algorithm for solving imperfect games. Basically it it a cross between deep_ctr and exploitability decent. The main idea behind it is to train a deep Policy network directly with exploitability decent. I made the implementation yesterday and well here it is:

https://github.com/dennisjay/open_spiel/blob/feature/nn_exploitablit_decent/open_spiel/python/algorithms/deep_exploitability_descent.py

This works very well for Kuhn poker and the exploitability reaches near nash equilibrum (<0.001). But for leduc poker convergence is very slow and gets stuck at approx. 1.3 . Do you have any tips how to improve my algorithm that I can contribute it to open_spiel.

Best regards
Dennis

**Comment by finbarrtimbers** [comment]
Unfortunately, that often happens, as Kuhn is so small that neural networks can often memorize it. 

If you write up the algorithm I'd be happy to take a look at it (ideally using LaTeX, maybe using [ShareLaTeX](https://www.sharelatex.com/)). 

I did notice that you're using vanilla SGD here, with a constant learning rate. In my experiments, I had to use a learning rate schedule to get it to converge. I found using Adam with `tf.train.exponential_decay` with a decay rate of 0.1 and a `decay_steps` value of 1000 worked for me. You are using a fairly high learning rate, so that could also be an issue.

**Comment by dennisjay** [comment]
Yeah tried some other optimizers like Adam and Momentum and a learning_rate decay but helps only a little... Still the convergence is much lower than I expected (compared to the results of optimizing the tabular game directly). I keep you updated.

**Comment by finbarrtimbers** [comment]
Please do!

**Comment by lanctot** [comment]
Any update @dennisjay ? Should we close this?

**Comment by dennisjay** [comment]
@lanctot yeah I’ll close it generally it seems to work but exploitability is going down very very slow. I don‘t have any other ideas to tune this. But I have a new idea stay tuned
